src/catalog.cc

```diff
@@ -79,16 +79,15 @@ void Catalog::Reset(OnSaveCallback callback) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
-bool Catalog::IsIdValid(const CatalogState& catalog_state) {
+bool Catalog::IsMatchingId(const CatalogState& catalog_state) {
   auto current_catalog_id = bundle_->GetCatalogId();
   auto new_catalog_id = catalog_state.catalog_id;
 
   if (current_catalog_id.empty()) {
-    // First time the catalog has been downloaded
-    return true;
+    // First time the catalog has been downloaded, so does not match
+    return false;
```

Comment:
can we add a test for this?

Reply:
Can we add unit tests once we have moved over to brave-core?